explorer/dockerExplorer.ts

```diff
@@ -17,20 +18,24 @@ export class DockerExplorerProvider implements vscode.TreeDataProvider<NodeBase>
     }
 
     refresh(): void {
+        docker.refreshEndpoint();
         this._onDidChangeTreeData.fire(this._imagesNode);
         this._onDidChangeTreeData.fire(this._containersNode);
         this._onDidChangeTreeData.fire(this._registriesNode);
     }
 
     refreshImages(): void {
+        docker.refreshEndpoint();
```

Comment:
Better would be to change Configuration.initialize in dockerExtension.ts to approximately this:

    export function initialize() {
        configurationListener = vscode.workspace.onDidChangeConfiguration((e: vscode.ConfigurationChangeEvent) => {
            // notify the language server that settings have change
            client.sendNotification(DidChangeConfigurationNotification.type, { settings: null });

            // Update endpoint if needed
            if (e.affectsConfiguration('docker')) {
                docker.refresh();
            }
        });
    }